tensorboard/webapp/widgets/line_chart_v2/line_chart_component.ts

```diff
@@ -226,10 +229,17 @@ export class LineChartComponent
       this.useDarkModeUpdated = true;
     }
 
-    if (this.scaleUpdated) {
+    // Do not set isViewBoxOverridden to false until updated userViewBox has been reflected on the lineChart.
+    if (this.scaleUpdated && !this.userViewBoxUpdated) {
```

Comment:
I haven't totally reasoned out why this is necessary but if you fully rely on the redux pattern to modify userViewBox (as I suggest in my other comment), then is it still necessary?

Reply:
I don't see how this.userViewBoxUpdated could be anything other than false here now, though. So I don't see how this if statement could ever evaluate to true.

You might set this.userViewBoxUpdated to true on L244 but then it will be immediately set back to false on L476 in the call to updateLineChart().

So it seems to me like L234 will never be executed.

Admittedly I haven't patched in the CL to give it a try -- I just wanted to check with you first whether there is something wrong with my reasoning.